docs/beyond_basics/101-181-metalad.rst

```diff
@@ -4,10 +4,363 @@ Metadata-Management with MetaLad
+     "path": "zen.png",
+     "dataset_id": "2d540a9d-2ef7-4b5f-8931-7c92f483f0c7",
+     "dataset_version": "19f2d98d758116d099d467260a5a71082b2c6a29",
+     "extractor_name": "Picture2Words",
+     "extractor_version": "0.1",
+     "extraction_parameter": {},
+     "extraction_time": 1675113291.1464975,
+     "agent_name": "Overworked CTO",
+     "agent_email": "closetoburnout@randomtechconsultancy.com",
+     "extracted_metadata": {
+     "description": "A lake with waterlilies in a front of snow covered mountains"
+	 }
+   }
+   EOT
+
+Then, we redirect the content of the file into the :command:`meta-add` command in the command line.
+The following call would add the metadata entry to the current dataset, ``cozy-screensavers``:
+
+
+.. runrecord:: _examples/DL-101-181-105
+   :language: console
+   :workdir: beyond_basics/meta/cozy-screensavers
+
+   $ datalad meta-add -d . - < metadata-zen.json
+
+.. find-out-more:: meta-add validity checks
+
+	When adding metadata for the first time, its not uncommon to run into errors.
+	Its quite easy, for example, to miss a comma or quotation mark when creating a JSON object by hand.
+	But there are also some internal checks that might be surprising.
+	If you want to add the metadata above to your own dataset, you should make sure to adjust the ``dataset_id`` to the ID of your own dataset, found via the command ``datalad configuration get datalad.dataset.id`` - otherwise you'll see an error [#f4]_, and likewise the ``dataset_version``.
+	And in case you'd supply the ``extraction_time`` as "this morning at 8AM" instead of a time stamp, the command will be unhappy as well.
+	In case an error occurs, make sure to read the error message, and turn the the commands' ``--help`` for insights about requirements you might have missed.
+
+After the metadata has been added, you can view it via the command :command:`meta-dump`.
+The simplest form of this command is ``meta-dump -r``, which will show all metadata that is stored in the dataset in the current directory.
+To get more specific metadata records, you can give a dataset-file-path-pattern to ``meta-dump``, much like an argument to ``ls``, that identifies :term:`dataset ID`, version and a file within the dataset.
+The two parts are separated by ``:``. The following line would just dump all metadata for ``zen.png``.
+
+.. runrecord:: _examples/DL-101-181-106
+   :language: console
+   :workdir: beyond_basics/meta/cozy-screensavers
+
+   $ datalad meta-dump -d . .:zen.png
+
+This could also be printed a bit more readable:
+
+.. runrecord:: _examples/DL-101-181-107
+   :language: console
+   :workdir: beyond_basics/meta/cozy-screensavers
+
+   $ datalad -f json_pp meta-dump -d . .:zen.png
+
+.. find-out-more:: More complex metadata-dumps
+
+   TODO: add complex Dataset-file-path-pattern examples, e.g., with UUIDs, versions, etc
+
+Using existing extractors to add metadata
+"""""""""""""""""""""""""""""""""""""""""
+
+TODO: This needs and example
```

Comment:
@christian-monch do you by chance have, e.g., the call how you used the study-mini-meta extractor? or shall I just cure the core extractor included in metalad here?

Reply:
Is this useful? I'm running it on a local version of the studyforrest superdataset which has a studyminimeta file added (see PR here: https://github.com/psychoinformatics-de/studyforrest-data/pull/61/files)

```bash
$ cd studyforrest-data
$ datalad meta-extract -d . metalad_studyminimeta > metadata_record.json

[INFO] Start studyminimeta metadata extraction from /Users/jsheunis/Documents/psyinf/Data/studyforrest-data
[INFO] Finished studyminimeta metadata extraction from /Users/jsheunis/Documents/psyinf/Data/studyforrest-data

$ cat extracted_metadata.json | jq .

{
  "type": "dataset",
  "dataset_id": "deabeb9b-7a37-4062-a1e0-8fcef7909609",
  "dataset_version": "6d7fb68264f9b9951ae141fc830712a8744e3293",
  "extractor_name": "metalad_studyminimeta",
  "extractor_version": "0.1",
  "extraction_parameter": {},
  "extraction_time": 1676280974.146493,
  "agent_name": "Stephan Heunis",
  "agent_email": "s.heunis@fz-juelich.de",
  "extracted_metadata": {
    "@context": {
      "@vocab": "http://schema.org/"
    },
    "@graph": [
      {
        "@id": "https://schema.datalad.org/datalad_dataset#deabeb9b-7a37-4062-a1e0-8fcef7909609",
        "@type": "Dataset",
        "version": "6d7fb68264f9b9951ae141fc830712a8744e3293",
        "name": "StudyForrest",
        "url": "https://github.com/psychoinformatics-de/studyforrest-data",
        "description": "The StudyForrest project centers around the use of the movie Forrest Gump, which provides complex sensory input that is both reproducible and is also richly laden with real-life-like content and contexts. Since its initial release, the StudyForrest dataset has grown and been extended substantially, and now encompasses many hours of fMRI scans, structural brain scans, eye-tracking data, and extensive annotations of the movie. It is a one-of-a-kind resource for studying high-level cognition in the human brain under complex, natural stimulation. The versatility of the provided data (some individuals have nearly ten hours of fMRI data) enables studies far beyond this main focus. This covers a vast range from studies of low-level signal properties and brain structure, to sensory integration and attentional processes, to computational modeling of representational spaces and brain area interactions.",
        "keywords": [
          "human",
          "fMRI",
          "task",
          "7T",
          "3T",
          "audio",
          "visual",
          "music",
          "retinotopy",
          "angiography",
          "T1,T2"
        ],
        "author": [
          {
            "@id": "https://schema.datalad.org/person#g@fz-juelich.de"
          }
        ],
        "funder": [
          {
            "@id": "https://schema.datalad.org/organization#BMBF, 01GQ1411",
            "@type": "Organization",
            "name": "BMBF, 01GQ1411"
          },
          {
            "@id": "https://schema.datalad.org/organization#NSF, 1429999",
            "@type": "Organization",
            "name": "NSF, 1429999"
          }
        ]
      },
      {
        "@id": "#publicationList",
        "@list": [
          {
            "@id": "#publication[0]",
            "@type": "ScholarlyArticle",
            "headline": "The effect of acquisition resolution on orientation decoding from V1: comparison of 3T and 7T",
            "datePublished": 2018,
            "sameAs": "https://doi.org/10.1101/305417",
            "author": [
              {
                "@id": "https://schema.datalad.org/person#a@fz-juelich.de"
              },
              {
                "@id": "https://schema.datalad.org/person#b@fz-juelich.de"
              },
              {
                "@id": "https://schema.datalad.org/person#c@fz-juelich.de"
              },
              {
                "@id": "https://schema.datalad.org/person#d@fz-juelich.de"
              },
              {
                "@id": "https://schema.datalad.org/person#e@fz-juelich.de"
              },
              {
                "@id": "https://schema.datalad.org/person#f@fz-juelich.de"
              },
              {
                "@id": "https://schema.datalad.org/person#g@fz-juelich.de"
              }
            ],
            "publication": {
              "@id": "https://schema.datalad.org/publication_event#bioRxiv",
              "@type": "PublicationEvent",
              "name": "bioRxiv"
            }
          },
          {
            "@id": "#publication[1]",
            "@type": "ScholarlyArticle",
            "headline": "Spatial band-pass filtering aids decoding musical genres from auditory cortex 7T fMRI",
            "datePublished": 2018,
            "sameAs": "https://doi.org/10.12688/f1000research.13689.2",
            "author": [
              {
                "@id": "https://schema.datalad.org/person#a@fz-juelich.de"
              },
              {
                "@id": "https://schema.datalad.org/person#f@fz-juelich.de"
              },
              {
                "@id": "https://schema.datalad.org/person#g@fz-juelich.de"
              }
            ],
            "publication": {
              "@id": "https://schema.datalad.org/publication_event#F1000Research",
              "@type": "PublicationEvent",
              "name": "F1000Research"
            }
          },
          {
            "@id": "#publication[2]",
            "@type": "ScholarlyArticle",
            "headline": "The effect of acquisition resolution on orientation decoding from V1 BOLD fMRI at 7 Tesla",
            "datePublished": 2017,
            "sameAs": "https://doi.org/10.1016/j.neuroimage.2016.12.040",
            "author": [
              {
                "@id": "https://schema.datalad.org/person#a@fz-juelich.de"
              },
              {
                "@id": "https://schema.datalad.org/person#c@fz-juelich.de"
              },
              {
                "@id": "https://schema.datalad.org/person#b@fz-juelich.de"
              },
              {
                "@id": "https://schema.datalad.org/person#f@fz-juelich.de"
              },
              {
                "@id": "https://schema.datalad.org/person#g@fz-juelich.de"
              }
            ],
            "publication": {
              "@id": "https://schema.datalad.org/publication_event#NeuroImage",
              "@type": "PublicationEvent",
              "name": "NeuroImage"
            }
          },
          {
            "@id": "#publication[3]",
            "@type": "ScholarlyArticle",
            "headline": "Ultra high-field multi-resolution fMRI data for orientation decoding in visual cortex",
            "datePublished": 2017,
            "sameAs": "https://doi.org/10.1016/j.dib.2017.05.014",
            "author": [
              {
                "@id": "https://schema.datalad.org/person#a@fz-juelich.de"
              },
              {
                "@id": "https://schema.datalad.org/person#b@fz-juelich.de"
              },
              {
                "@id": "https://schema.datalad.org/person#c@fz-juelich.de"
              },
              {
                "@id": "https://schema.datalad.org/person#d@fz-juelich.de"
              },
              {
                "@id": "https://schema.datalad.org/person#e@fz-juelich.de"
              },
              {
                "@id": "https://schema.datalad.org/person#f@fz-juelich.de"
              },
              {
                "@id": "https://schema.datalad.org/person#g@fz-juelich.de"
              }
            ],
            "publication": {
              "@id": "https://schema.datalad.org/publication_event#Data in Brief",
              "@type": "PublicationEvent",
              "name": "Data in Brief"
            }
          }
        ]
      },
      {
        "@id": "#personList",
        "@list": [
          {
            "@id": "https://schema.datalad.org/person#a@fz-juelich.de",
            "@type": "Person",
            "email": "a@fz-juelich.de",
            "name": " Ayan Sengupta",
            "givenName": "Ayan",
            "familyName": "Sengupta"
          },
          {
            "@id": "https://schema.datalad.org/person#b@fz-juelich.de",
            "@type": "Person",
            "email": "b@fz-juelich.de",
            "name": " Oliver Speck",
            "givenName": "Oliver",
            "familyName": "Speck"
          },
          {
            "@id": "https://schema.datalad.org/person#c@fz-juelich.de",
            "@type": "Person",
            "email": "c@fz-juelich.de",
            "name": " Renat Yakupov",
            "givenName": "Renat",
            "familyName": "Yakupov"
          },
          {
            "@id": "https://schema.datalad.org/person#d@fz-juelich.de",
            "@type": "Person",
            "email": "d@fz-juelich.de",
            "name": " Martin Kanowski",
            "givenName": "Martin",
            "familyName": "Kanowski"
          },
          {
            "@id": "https://schema.datalad.org/person#e@fz-juelich.de",
            "@type": "Person",
            "email": "e@fz-juelich.de",
            "name": " Claus Tempelmann",
            "givenName": "Claus",
            "familyName": "Tempelmann"
          },
          {
            "@id": "https://schema.datalad.org/person#f@fz-juelich.de",
            "@type": "Person",
            "email": "f@fz-juelich.de",
            "name": " Stefan Pollmann",
            "givenName": "Stefan",
            "familyName": "Pollmann"
          },
          {
            "@id": "https://schema.datalad.org/person#g@fz-juelich.de",
            "@type": "Person",
            "email": "g@fz-juelich.de",
            "name": " Michael Hanke",
            "givenName": "Michael",
            "familyName": "Hanke",
            "sameAs": "0000-0001-6398-6370"
          }
        ]
      }
    ]
  }
}

$ datalad meta-add -d . metadata_record.json

meta_add(ok): /Users/jsheunis/Documents/psyinf/Data/studyforrest-data (dataset) [added dataset metadata to /Users/jsheunis/Documents/psyinf/Data/studyforrest-data]
```

Reply:
I used ``meta-conduct`` in a superset to extract metadata from all subdatasets and add it to the git-repo (aka metadata-store) of the superdataset. I think that might be too complex here, since we did not yet describe ``meta-conduct``. For completeness, this was the commands I used:

```shell
> cd superset
> datalad meta-conduct extract_metadata traverser.top_level_dir=$(pwd) traverser.item_type=dataset traverser.traverse_sub_datasets=True extractor.extractor_type=dataset extractor.extractor_name=metalad_studyminimeta adder.aggregate=True
```

If you just want to use the extractor on a specific dataset, you can execute:
```
> datalad meta-extract -d <path-to-dataset> metalad_studyminimeta
``` 
But that would require a ``.studyminimeta.yaml`` file in the dataset. Therefore it might indeed be better to use the core extractor. That would be done with a simple command:

```shell
> datalad meta-extract -d <path-to-dataset> metalad_core
```

or with 

```shell
> datalad meta-extract metalad_core
```
if the current working directory is the root of the dataset from which you want to extract core data.

Reply:
thx, I think I added a very simple example with metalad_core